src/erlfmt.erl

```diff
@@ -151,26 +151,40 @@ format_string_full(String, Options) ->
         {error, Error} -> {error, Error}
     end.
 
-contains_pragma_node(Node) ->
+which_pragma_node(Node) ->
     {PreComments, _, PostComments} = erlfmt_format:comments_with_pre_dot(Node),
-    lists:any(fun contains_pragma_comment/1, PreComments ++ PostComments).
+    Comments = PreComments ++ PostComments,
+    case lists:any(fun contains_noformat_pragma/1, Comments) of
+        true ->
+            noformat;
+        false ->
+            case lists:any(fun contains_format_pragma/1, Comments) of
+                true -> format;
+                false -> nopragma
+            end
+    end.
 
-contains_pragma_comment({comment, _Loc, Comments}) ->
+contains_format_pragma({comment, _Loc, Comments}) ->
     string:find(Comments, "@format") =/= nomatch.
 
+contains_noformat_pragma({comment, _Loc, Comments}) ->
+    string:find(Comments, "@noformat") =/= nomatch.
+
 %% insert_pragma_nodes only inserts an @format comment,
 %% if one has not already been inserted.
 insert_pragma_nodes([]) ->
     [];
 insert_pragma_nodes([{shebang, _, _} = Node | Nodes]) ->
-    case contains_pragma_node(Node) of
-        true -> [replace_pragma_node(Node) | Nodes];
-        false -> [Node | insert_pragma_nodes(Nodes)]
+    case which_pragma_node(Node) of
+        format -> [replace_pragma_node(Node) | Nodes];
+        noformat -> [Node | Nodes];
+        nopragma -> [Node | insert_pragma_nodes(Nodes)]
     end;
 insert_pragma_nodes([Node | Nodes]) ->
-    case contains_pragma_node(Node) of
-        true -> [replace_pragma_node(Node) | Nodes];
-        false -> [insert_pragma_node(Node) | Nodes]
+    case which_pragma_node(Node) of
+        format -> [replace_pragma_node(Node) | Nodes];
+        noformat -> [Node | Nodes];
+        nopragma -> [insert_pragma_node(Node) | Nodes]
```

Comment:
I really like how those matches look :+1:

Reply:
Thank you :)